staking/contracts/contracts.py

```diff
@@ -76,44 +101,48 @@ def send_asset(
 @Subroutine(TealType.none)
 def is_not_paused() -> Expr:
     return Seq(
-        Assert(Not(App.globalGet(Bytes("P")))),
+        Assert(Not(App.globalGet(PAUSED_FLAG))),
     )
 
 @Subroutine(TealType.none)
 def calculate_rewards(addr: Expr) -> Expr:
     return Seq(
         # Skip if not begun
-        If(Global.latest_timestamp() > App.globalGet(Bytes("BT")), Return()),
+        If(Global.latest_timestamp() < App.globalGet(BEGIN_TIMESTAMP), Return()),
 
         # Skip if updated since ET
-        If(App.localGet(addr, Bytes("LU")) < App.globalGet(Bytes("ET")), Return()),
+        If(App.localGet(addr, LAST_UPDATED) > App.globalGet(END_TIMESTAMP), Return()), 
 
         # Calculate time since last update
         # End
         (end := ScratchVar()).store(
-            If(Global.latest_timestamp() > App.globalGet(Bytes("ET")))
-            .Then(App.globalGet(Bytes("ET")))
+            If(Global.latest_timestamp() > App.globalGet(END_TIMESTAMP))
+            .Then(App.globalGet(END_TIMESTAMP))
             .Else(Global.latest_timestamp())
         ),
         # Start
         (start := ScratchVar()).store(
-            If(App.localGet(addr, Bytes("LU")) < App.globalGet(Bytes("BT")))
-            .Then(App.globalGet(Bytes("BT")))
-            .Else(App.localGet(addr, Bytes("LU")))
+            If(App.localGet(addr, LAST_UPDATED) < App.globalGet(BEGIN_TIMESTAMP))
+            .Then(App.globalGet(BEGIN_TIMESTAMP))
+            .Else(App.localGet(addr, LAST_UPDATED))
         ),
         # Duration
         (duration := ScratchVar()).store(end.load() - start.load()),
 
-        # Calculate time since last updated
+        # Calculate rewards
         (rewards := ScratchVar()).store(
-            App.localGet(addr, Bytes("AS")) * duration.load() / Int(31557600) * App.globalGet(Bytes("FR")) / Int(10000)
+            App.localGet(addr, AMOUNT_STAKED) * duration.load() / Int(31557600) * App.globalGet(FIXED_RATE) / Int(10000) # I think there might be truncation issues, especially with very high denominator values.
         ),
 
         # Remove rewards from global
-        App.globalPut(Bytes("TR"), App.globalGet(Bytes("TR")) - rewards.load()),
+        App.globalPut(TOTAL_REWARDS, App.globalGet(TOTAL_REWARDS) - rewards.load()), # What happens after "TR" runs out and goes to zero? I think the transaction will fail and any method (i.e. deposit, withdraw) that wants to calculate rewards will also fail. Would anybody be able to take out assets if this happens?
```

Comment:
That's a great find! I have just updated it to check there's enough rewards to calculate it, otherwise only reward whatever is left (if anything at all). So now people can continue to withdraw their stake once the rewards have been depleted.